rocAL/rocAL/source/api/rocal_api_data_loaders.cpp

```diff
@@ -237,12 +238,13 @@ rocalJpegFileSource(
 
         INFO("Internal buffer size width = "+ TOSTR(width)+ " height = "+ TOSTR(height) + " depth = "+ TOSTR(num_of_planes))
 
-        auto info = ImageInfo(width, height,
-                              context->internal_batch_size(),
-                              num_of_planes,
-                              context->master_graph->mem_type(),
-                              color_format );
-        output = context->master_graph->create_loader_output_image(info);
+        std::vector<size_t> dims = {context->user_batch_size(), height, width, num_of_planes};
+        auto info  = rocalTensorInfo(std::vector<size_t>(std::move(dims)),
```

Comment:
Why have we done a typecasting here for std::move(dims). Is there a specific reason ?
Please check if this can be done without a typecast